src/SlashCommands.tsx

```diff
@@ -141,13 +141,25 @@ export class Command {
 
     run(roomId: string, threadId: string, args: string) {
         // if it has no runFn then its an ignored/nop command (autocomplete only) e.g `/me`
-        if (!this.runFn) return reject(_t("Command error"));
+        if (!this.runFn) {
+            return reject(
+                newTranslatableError(
```

Comment:
`newTranslatableError` don't get picked up by gen-i18n, it only looks at `_t` and `_td` atm, so won't detect when we are missing en translations. We either need to use `_td` within `newTranslatableError` calls or add `newTranslatableError` to gen-i18n

Reply:
See how many missing strings there are in the i18n files now.

Reply:
> See how many missing strings there are in the i18n files now.

😱 Makes sense, good spot

>  or add `newTranslatableError` to gen-i18n

I think I am leaning to this as it's probably useful to others as well to have this utility for all projects,

https://github.com/matrix-org/matrix-web-i18n/blob/e5c7071e0cdf715de87ef39dc8260e11d7add2f8/scripts/gen-i18n.js#L35

Reply:
Is something like this workable? https://github.com/matrix-org/matrix-web-i18n/pull/3

That we can manually configure whatever translation functions we want per project as they come up.

Reply:
Yeah maybe, but might be strange to need to define it in matrix-react-sdk's package.json as well as element-web's package.json

Reply:
It looks like we check-in the translations for `matrix-react-sdk`. Do we recollect them from `element-web` instead of just pulling them in? When in `element-web`, does it look across all `node_modules`?

---

Thinking about similar recursive configs out there but these go from current directory inheriting outwards.

 - https://github.com/davidtheclark/cosmiconfig (what `stylelint` uses)
 - eslint: https://github.com/eslint/eslint/blob/dd58cd4afa6ced9016c091fc99a702c97a3e44f0/lib/init/npm-utils.js#L21-L40
 - jest
 
---

Otherwise, I can just hardcode `newTranslatableError` in there 🤷‍♀️ (like https://github.com/matrix-org/matrix-web-i18n/pull/2)

Reply:
element-web layer only collects its own translations which are then merged with node_modules/matrix-react-sdk/src/i18n

Reply:
> element-web layer only collects its own translations which are then merged with node_modules/matrix-react-sdk/src/i18n

In that case, seems reasonable to have it in both projects since it's used in both. Is this solution good enough for this to merge once we get https://github.com/matrix-org/matrix-web-i18n/pull/3 in?

If it was a transient dependency where we collect from `node_modules/matrix-react-sdk`, I would be less keen on it.

Reply:
Sure

Reply:
https://github.com/matrix-org/matrix-web-i18n/pull/3 is now merged 🚀

`newTranslatableError` is now specified as a translatable func (see `matrix_i18n_extra_translation_funcs` in the `package.json`)